In [40]:
import pandas as pd
import numpy as np


In [59]:
# Loading the original datafile for google play
orginal_play = pd.read_csv('googleplaystore.csv')
original_review = pd.read_csv('googleplaystore_user_reviews.csv')

# Making a copy of the file
play = orginal_play.copy()
review = original_review.copy()

In [63]:
review = review.dropna()
review.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 37427 entries, 0 to 64230
Data columns (total 5 columns):
App                       37427 non-null object
Translated_Review         37427 non-null object
Sentiment                 37427 non-null object
Sentiment_Polarity        37427 non-null float64
Sentiment_Subjectivity    37427 non-null float64
dtypes: float64(2), object(3)
memory usage: 1.7+ MB


In [42]:
#Checking data
play.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [67]:
review

App  \
0               10 Best Foods for You   
1               10 Best Foods for You   
3               10 Best Foods for You   
4               10 Best Foods for You   
5               10 Best Foods for You   
6               10 Best Foods for You   
8               10 Best Foods for You   
9               10 Best Foods for You   
10              10 Best Foods for You   
11              10 Best Foods for You   
12              10 Best Foods for You   
13              10 Best Foods for You   
14              10 Best Foods for You   
16              10 Best Foods for You   
17              10 Best Foods for You   
18              10 Best Foods for You   
19              10 Best Foods for You   
20              10 Best Foods for You   
21              10 Best Foods for You   
22              10 Best Foods for You   
23              10 Best Foods for You   
24              10 Best Foods for You   
25              10 Best Foods for You   
26              10 Best Foods for You   
27              10 Best Foods for You   
28              10 Best Foods for You   
29              10 Best Foods for You   
30              10 Best Foods for You   
31              10 Best Foods for You   
32              10 Best Foods for You   
...                               ...   
64185  Hotwire Hotel & Car Rental App   
64186  Hotwire Hotel & Car Rental App   
64187  Hotwire Hotel & Car Rental App   
64188  Hotwire Hotel & Car Rental App   
64189  Hotwire Hotel & Car Rental App   
64190  Hotwire Hotel & Car Rental App   
64191  Hotwire Hotel & Car Rental App   
64192  Hotwire Hotel & Car Rental App   
64194  Hotwire Hotel & Car Rental App   
64196  Housing-Real Estate & Property   
64197  Housing-Real Estate & Property   
64198  Housing-Real Estate & Property   
64199  Housing-Real Estate & Property   
64200  Housing-Real Estate & Property   
64201  Housing-Real Estate & Property   
64205  Housing-Real Estate & Property   
64207  Housing-Real Estate & Property   
64209  Housing-Real Estate & Property   
64213  Housing-Real Estate & Property   
64215  Housing-Real Estate & Property   
64216  Housing-Real Estate & Property   
64217  Housing-Real Estate & Property   
64218  Housing-Real Estate & Property   
64220  Housing-Real Estate & Property   
64221  Housing-Real Estate & Property   
64222  Housing-Real Estate & Property   
64223  Housing-Real Estate & Property   
64226  Housing-Real Estate & Property   
64227  Housing-Real Estate & Property   
64230  Housing-Real Estate & Property   

                                       Translated_Review Sentiment  \
0      I like eat delicious food. That's I'm cooking ...  Positive   
1        This help eating healthy exercise regular basis  Positive   
3             Works great especially going grocery store  Positive   
4                                           Best idea us  Positive   
5                                               Best way  Positive   
6                                                Amazing  Positive   
8                                   Looking forward app,   Neutral   
9                  It helpful site ! It help foods get !   Neutral   
10                                             good you.  Positive   
11     Useful information The amount spelling errors ...  Positive   
12     Thank you! Great app!! Add arthritis, eyes, im...  Positive   
13     Greatest ever Completely awesome maintain heal...  Positive   
14     Good health...... Good health first priority.....  Positive   
16     Health It's important world either life . thin...  Positive   
17     Mrs sunita bhati I thankful developers,to make...  Positive   
18     Very Useful in diabetes age 30. I need control...  Positive   
19                                    One greatest apps.  Positive   
20                                             good nice  Positive   
21                                 Healthy Really helped  Positive   
22                                            God health   Neutral   
23     HEALTH SHOULD 

In [43]:
play.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
App               10841 non-null object
Category          10841 non-null object
Rating            9367 non-null float64
Reviews           10841 non-null object
Size              10841 non-null object
Installs          10841 non-null object
Type              10840 non-null object
Price             10841 non-null object
Content Rating    10840 non-null object
Genres            10841 non-null object
Last Updated      10841 non-null object
Current Ver       10833 non-null object
Android Ver       10838 non-null object
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


Total rows 10841

### Quality
* Life Made WI-Fi Touchscreen shifted, data needs to be adjusted
* Numeric day of the week is not required, so can be removed for analysis

### Tidiness
* Installs has plus symbol that needs to be removed
* Android ver needs to remove and up
* M removed from size
* Reviews, installs, price need to change to floats(int)
* Last updated changed to datetime
* Change Category to title case
* Remove _ from Category


## Incomplete

* Possibly split up Genres for games with multiple genres
* CC Rivals, Type needs to be changed to Free
* Convert size to KB and then float

Current Version Changes
1.1.2 - Learn To Draw Kawaii Characters
4 - Virtual DJ Sound Mixer
1.1.2 - BT Master	
2.17.8 - Dots puzzle

Group Data into categories i.e. low, med, high
Price
Rating



## Define
The titles of the files have spaces in them, these will need to be changed to add an underscore in place of those spaces

## Code

In [44]:
play = play.rename(columns = {'Content Rating': 'Content_Rating', 'Last Updated': 'Last_Updated', 'Current Ver': 'Current_Ver', 'Android Ver': 'Android_Ver'})

## Test

In [45]:
play.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content_Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last_Updated         Current_Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android_Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [46]:
play['Current_Ver'] = play['Current_Ver'].str.replace('4.0 and up', '1.0.19')
play['Last_Updated'] = play['Last_Updated'].str.replace('1.0.19', 'February 11, 2018')
play['Genres'] = play['Genres'].str.replace('February 11, 2018', 'Lifestyle')
play['Content_Rating'] = play['Content_Rating'].fillna('Everyone')
play['Price'] = play['Price'].str.replace('Everyone', 'Free')
play['Type'] = play['Type'].str.replace('0', 'Free')
play['Installs'] = play['Installs'].str.replace('Free', '1000')
play['Size'] = play['Size'].str.replace('1,000+', '3')
play['Size'] = play['Size'].str.replace('+', '')
play['Reviews'] = play['Reviews'].str.replace('3.0M', '19')
play['Rating'] = play['Rating'].replace(19,1.9)
play['Category'] = play['Category'].str.replace('1.9', 'LIFESTYLE')
play[play['App']== 'Life Made WI-Fi Touchscreen Photo Frame']



App   Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame  LIFESTYLE     1.9      19   

      Size Installs  Type Price Content_Rating     Genres       Last_Updated  \
10472    3     1000  Free  Free       Everyone  Lifestyle  February 11, 2018   

      Current_Ver Android_Ver  
10472      1.0.19         NaN

## Define:
Several of the columns have values that are not required. The + sign from Installs, the 'and up' from Android Ver, the 'M' from Size. This section makes those changes

In [47]:
#Removing the Plus sign from installs
play['Installs'] = play['Installs'].map(lambda x: x.rstrip('+')).str.replace(',','')

#Removing the 'and up' from Android Version
play['Android_Ver'] = play['Android_Ver'].str.replace('and up','')

play['Size'] = play['Size'].str.replace('M','')

play['Price'] = play['Price'].str.replace('$','')
play['Price'] = play['Price'].str.replace('Free','0')

play.tail()

App             Category  \
10836                               Sya9a Maroc - FR               FAMILY   
10837               Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                         Parkinson Exercices FR              MEDICAL   
10839                  The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840  iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size  Installs  Type Price  \
10836     4.5      38                  53      5000  Free     0   
10837     5.0       4                 3.6       100  Free     0   
10838     NaN       3                 9.5      1000  Free     0   
10839     4.5     114  Varies with device      1000  Free     0   
10840     4.5  398307                  19  10000000  Free     0   

      Content_Rating             Genres      Last_Updated         Current_Ver  \
10836       Everyone          Education     July 25, 2017                1.48   
10837       Everyone          Education      July 6, 2018                 1.0   
10838       Everyone            Medical  January 20, 2017                 1.0   
10839     Mature 17+  Books & Reference  January 19, 2015  Varies with device   
10840       Everyone          Lifestyle     July 25, 2018  Varies with device   

              Android_Ver  
10836                4.1   
10837                4.1   
10838                2.2   
10839  Varies with device  
10840  Varies with device

## Define
Converting datatypes for Reviews, Installs, Price, and Size

## Code

In [48]:
convert_dict = {'Reviews': int, 'Installs': int, 'Price':float}
play = play.astype(convert_dict)
play['Last_Updated'] = pd.to_datetime(play.Last_Updated)
play['Month'] = play['Last_Updated'].dt.strftime('%m')
play['Last_Updated'] = play['Last_Updated'].dt.strftime('%m/%y')


In [49]:
play

App             Category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
5                             Paper flowers instructions       ART_AND_DESIGN   
6                Smoke Effect Photo Maker - Smoke Editor       ART_AND_DESIGN   
7                                       Infinite Painter       ART_AND_DESIGN   
8                                   Garden Coloring Book       ART_AND_DESIGN   
9                          Kids Paint Free - Drawing Fun       ART_AND_DESIGN   
10                               Text on Photo - Fonteee       ART_AND_DESIGN   
11               Name Art Photo Editor - Focus n Filters       ART_AND_DESIGN   
12                        Tattoo Name On My Photo Editor       ART_AND_DESIGN   
13                                 Mandala Coloring Book       ART_AND_DESIGN   
14       3D Color Pixel by Number - Sandbox Art Coloring       ART_AND_DESIGN   
15                       Learn To Draw Kawaii Characters       ART_AND_DESIGN   
16          Photo Designer - Write your name with shapes       ART_AND_DESIGN   
17                              350 Diy Room Decor Ideas       ART_AND_DESIGN   
18                         FlipaClip - Cartoon animation       ART_AND_DESIGN   
19                                          ibis Paint X       ART_AND_DESIGN   
20                           Logo Maker - Small Business       ART_AND_DESIGN   
21             Boys Photo Editor - Six Pack & Men's Suit       ART_AND_DESIGN   
22               Superheroes Wallpapers | 4K Backgrounds       ART_AND_DESIGN   
23                                Mcqueen Coloring pages       ART_AND_DESIGN   
24                           HD Mickey Minnie Wallpapers       ART_AND_DESIGN   
25                            Harley Quinn wallpapers HD       ART_AND_DESIGN   
26                         Colorfit - Drawing & Coloring       ART_AND_DESIGN   
27                                 Animated Photo Editor       ART_AND_DESIGN   
28                                 Pencil Sketch Drawing       ART_AND_DESIGN   
29                       Easy Realistic Drawing Tutorial       ART_AND_DESIGN   
...                                                  ...                  ...   
10811                                        FR Plus 1.6    AUTO_AND_VEHICLES   
10812                                      Fr Agnel Pune               FAMILY   
10813                                     DICT.fr Mobile             BUSINESS   
10814                               FR: My Secret Pets!                FAMILY   
10815                          Golden Dictionary (FR-AR)  BOOKS_AND_REFERENCE   
10816                                 FieldBi FR Offline             BUSINESS   
10817                               HTC Sense Input - FR                TOOLS   
10818                               Gold Quote - Gold.fr              FINANCE   
10819                                          Fanfic-FR  BOOKS_AND_REFERENCE   
10820                                    Fr. Daoud Lamei               FAMILY   
10821                                            Poop FR               FAMILY   
10822                                          PLMGSS FR         PRODUCTIVITY   
10823                                       List iptv FR        VIDEO_PLAYERS   
10824                                          Cardio-FR              MEDICAL   
10825                                 Naruto & Boruto FR               SOCIAL   
10826          Frim: get new friends on local chat rooms               SOCIAL   
10827                                 Fr Agnel Ambarnath               FAMILY   
10828                            Manga-FR - Anime Vostfr               COMICS   
108

## Test


In [50]:
play.dtypes

App                object
Category           object
Rating            float64
Reviews             int32
Size               object
Installs            int32
Type               object
Price             float64
Content_Rating     object
Genres             object
Last_Updated       object
Current_Ver        object
Android_Ver        object
Month              object
dtype: object

## Define
The titles in categories needs to be changed to title case, the _ also needs to be removed

## CODE

In [51]:
play['Category'] = play['Category'].str.title()
play['Category'] = play['Category'].str.replace('And','and')
play['Category'] = play['Category'].str.replace('_' , ' ')
play['App'] = play['App'].str.title()


In [52]:
play.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & Scrapbook  Art and Design     4.1   
1                                Coloring Book Moana  Art and Design     3.9   
2  U Launcher Lite – Free Live Cool Themes, Hide ...  Art and Design     4.7   
3                              Sketch - Draw & Paint  Art and Design     4.5   
4              Pixel Draw - Number Art Coloring Book  Art and Design     4.3   

   Reviews Size  Installs  Type  Price Content_Rating  \
0      159   19     10000  Free    0.0       Everyone   
1      967   14    500000  Free    0.0       Everyone   
2    87510  8.7   5000000  Free    0.0       Everyone   
3   215644   25  50000000  Free    0.0           Teen   
4      967  2.8    100000  Free    0.0       Everyone   

                      Genres Last_Updated         Current_Ver Android_Ver  \
0               Art & Design        01/18               1.0.0      4.0.3    
1  Art & Design;Pretend Play        01/18               2.0.0      4.0.3    
2               Art & Design        08/18               1.2.4      4.0.3    
3               Art & Design        06/18  Varies with device        4.2    
4    Art & Design;Creativity        06/18                 1.1        4.4    

  Month  
0    01  
1    01  
2    08  
3    06  
4    06

In [53]:
play = play.drop(['Current_Ver', 'Android_Ver'], axis=1)

In [39]:
export = play.to_csv('play_clean.csv', index=None, header=True)